### PR TITLE
Support variable drip intervals (day/hour/minute/second)

### PR DESCRIPTION
**Pull request by pjeby**
## Description

In certain circumstances, it's useful to enforce a drip delay that's less than a day, e.g. to help students avoid the temptation to skip ahead and not do an (offline) exercise, or to force spacing between repetitions.  This PR adds support for shorter drip intervals, by adding a dropdown to select the interval multiplier (days/hours/minutes/seconds).

## How has this been tested?
* Manual testing in Chrome
* Modifications made to existing unit tests for existing drip delay ( `get_available_date()`, `is_available()`) to cover both the default case of days, and non-default intervals such as hours.

### As Seen In Metabox
Metabox

### As Seen In Course Builder
Course Builder

## Types of changes

This is a new feature, introducing a lesson meta field `drip_multiplier`, which defaults to 86400 (one day), replacing the `DAY_IN_SECONDS` constant previously used in `get_available_date()`.  (The default ensures backward compatibility for existing lessons.)  A dropdown was added to both the lesson builder schema and lesson metabox to support choosing the multiplier.  i18n should not be affected, since only existing strings were used.

Because of changes to JavaScript and SCSS, a selective asset rebuild was required (`gulp styles js:builder`), but for ease of code review and to avoid spurious commit conflicts I have not added the built assets to this PR at this time.  (If you would prefer them included, however, just let me know and I'll push them as well.).

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding Standards.

### CODE REVIEW

**Comment by thomasplevy** [comment]
@pjeby thanks for this!!! At a glance it looks really solid but I'm going to pull it and down and play around with it before merging. It looks well thought out and all the initial gotchas I can think of seem to have already been considered so it should be good to go.

I'll let you know if I spot anything or see any issues.

Thanks again

**Comment by thomasplevy** [comment]
@actual-saurabh could you provide me with a second set of eyes on this please

File: assets/js/builder/Schemas/Lesson.js [hunk 1]
```diff
@@ -134,10 +134,39 @@ define( [], function() {
...
+								{
+									key: '86400',
+									val: LLMS.l10n.translate( 'days' ),
+								},
+								{
+									key: '3600',
+									val: LLMS.l10n.translate( 'hours' ),
+								},
+								{
+									key: '60',
```
Comment (actual-saurabh) [inline assets/js/builder/Schemas/Lesson.js]: Change key as above

File: assets/js/builder/Schemas/Lesson.js [hunk 2]
```diff
@@ -134,10 +134,39 @@ define( [], function() {
...
+									val: LLMS.l10n.translate( 'minutes' ),
+								},
+								{
+									key: '1',
```
Comment (actual-saurabh) [inline assets/js/builder/Schemas/Lesson.js]: change key as above

**Review by actual-saurabh — CHANGES_REQUESTED** [review]
The naming of Drip Multipliers, etc is confusing and doesn't match user mental models. Please use Interval and Units as described in details in the comments

**Comment by thomasplevy** [comment]
@pjeby,

Thanks for participating in our [LifterLMS contributor month hacktoberfest celebration](https://make.lifterlms.com/2018/09/26/hacktoberfest/)!

If you'd head over to https://lifterlms.com/contact and send us your shipping address we'll get your participation reward in the mail.

Thanks,

**Comment by pjeby** [comment]
Thanks.  Is there anything else I can to do to help move this forward?

**Comment by pjeby** [comment]
Not a problem; I just wanted to make sure we weren't both waiting for the other party to do something.  :wink:

**Comment by pjeby** [comment]
So, what was the result of the review?  Was there an issue with the implementation, or have you just decided this isn't a feature you want?

**Comment by thomasplevy** [comment]
@pjeby My apologies, I was cleaning up some invalid PRs yesteday and I bulk closed 4 or 5, I accidentally closed this one. 

I'm sorry this has been pending for so long. We're a small team and we have a lot of things to work on and I just haven't been able to prioritize this as it's not something we really need at this moment based on requests from our users.

My apologies!

**Comment by pjeby** [comment]
So, was this a bulk closure again, or...?

**Comment by thomasplevy** [comment]
@pjeby my apologies... we've undertaken a project of renaming branches across all our repos and I didn't realize it would close open PRs

If you could repopen this against the "dev" branch we'll review when time allows.

**Pull request closed without merging**